src/http-adapter.js

```diff
@@ -76,32 +80,39 @@ class HttpAdapter {
    * @returns {Promise} Promise for destroy.
    */
   destroy (link) {
-    return this._ajax('DELETE', link)
+    return this._http(METHOD.DELETE, link)
   }
 
-  _ajax (type, url, data) {
-    let headers = {
-      'Accept': 'application/vnd.api+json',
-      'Content-Type': 'application/vnd.api+json'
+  async _http (method = '', url = '', data = {}) {
+    let init = {
+      method,
+      headers: {
+        'Accept': 'application/vnd.api+json',
+        'Content-Type': 'application/vnd.api+json'
+      }
     }
 
-    // Stringify data before any async stuff, just in case it's accidentally a mutable object (e.g.
-    // some instrumented Vue data)
-    if (data && ['PATCH', 'POST'].indexOf(type) > -1) {
-      data = JSON.stringify(data)
+    url = new URL(url, location.origin)
+
+    if (method.includes(METHOD.GET)) {
+      // serialize query params
+      Object.keys(data).forEach(key => url.searchParams.append(key, data[key]))
+    } else if (method.includes(METHOD.POST) || method.includes(METHOD.PATCH)) {
+      // Stringify data before any async stuff, just in case it's accidentally a mutable object (e.g.
+      // some instrumented Vue data)
+      init.body = JSON.stringify(data)
     }
 
     let promise
     if (this.serializeRequests) {
       // Wait for all requests to settle (either with success or rejection) before making request
-      const promises = Array.from(this._outstandingRequests)
-        .map(promise => promise.catch(() => {}))
+      const promises = [...this._outstandingRequests].map(promise =>
+        promise.catch(() => {})
+      )
 
-      promise = Promise.all(promises)
-        .then(() => this._makeRequest({ url, type, headers, data }))
-    } else {
-      promise = this._makeRequest({ url, type, headers, data })
+      await Promise.all(promises)
```

Comment:
This breaks the request serialization mechanism; under the previous model requests that are waiting would be added to `_outstandingRequests` while they are waiting for previous requests. With this model, requests that were started while waiting for a previous request would then run in parallel, not in sequence.